frontend/src/js/components/modals/TemplateEditor.jsx

```diff
@@ -1,10 +1,15 @@
 import React from 'react'
 import TemplatePreview from '../TemplatePreview.jsx'
+import TemplateLabelsEditor from '../editors/template/TemplateLabelsEditor.jsx'
 
 
 var TemplateEditor = React.createClass({
     propTypes: {
-        template: React.PropTypes.object
+        template: React.PropTypes.object,
+        updateTemplate: React.PropTypes.func,
+    },
+    onUpdate(attribute, key, value) {
+        this.props.updateTemplate(this.props.template.name, attribute, key, value);
```

Comment:
We should update this to use the React `state` system to have an internal template that we're editing, and then a "Save" button that syncs the internal template with the external one (so we can have a "cancel" button that resets everything appropriately by *not* syncing and just killing the modal).

The code you need to add looks something like this:

```
getInitialState() {
    return {workingTemplate: $.extend({}, this.props.template)};
},
componentWillReceiveProps(nextProps) {
    this.setState({workingTemplate: $.extend({}, nextProps.template)});
}
```

and then using `this.state.workingTemplate` where we're using `this.props.template` currently.

You can come to me with questions about that.